File: SystemUIShared/src/com/android/systemui/shared/QuickstepCompat.java
package com.android.systemui.shared;

import android.annotation.SuppressLint;
import android.os.Build;

import app.lawnchair.compatlib.ActivityManagerCompat;
import app.lawnchair.compatlib.QuickstepCompatFactory;
import app.lawnchair.compatlib.eleven.QuickstepCompatFactoryVR;
import app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS;

public class QuickstepCompat {

    private static final QuickstepCompatFactory sFactory;
    private static final ActivityManagerCompat sActivityManagerCompat;

    @SuppressLint("AnnotateVersionCheck")
    public static final boolean ATLEAST_S = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S;

    static {
        if (ATLEAST_S) {
            sFactory = new QuickstepCompatFactoryVS();
        } else {
            sFactory = new QuickstepCompatFactoryVR();
        }
        sActivityManagerCompat = sFactory.getActivityManagerCompat();
    }

    public static QuickstepCompatFactory getFactory() {
        return sFactory;
    }

    public static ActivityManagerCompat getActivityManagerCompat() {
        return sActivityManagerCompat;
    }
}

File: systemUIShared/src/app/lawnchair/compat/QuickstepCompat.kt
package app.lawnchair.compat

import android.os.Build
import app.lawnchair.compatlib.ActivityManagerCompat
import app.lawnchair.compatlib.QuickstepCompatFactory
import app.lawnchair.compatlib.eleven.QuickstepCompatFactoryVR
import app.lawnchair.compatlib.ten.QuickstepCompatFactoryVQ
import app.lawnchair.compatlib.thirteen.QuickstepCompatFactoryVT
import app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS

object QuickstepCompat {
    @JvmField
    val ATLEAST_R: Boolean = Build.VERSION.SDK_INT >= Build.VERSION_CODES.R

    @JvmField
    val ATLEAST_S: Boolean = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S

    @JvmField
    val ATLEAST_T: Boolean = Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU


    @JvmStatic
    val factory: QuickstepCompatFactory = if (ATLEAST_T) {
        QuickstepCompatFactoryVT()
    } else if (ATLEAST_S) {
        QuickstepCompatFactoryVS()
    } else if (ATLEAST_R) {
        QuickstepCompatFactoryVR()
    } else {
        QuickstepCompatFactoryVQ()
    }

    @JvmStatic
    val activityManagerCompat: ActivityManagerCompat = factory.activityManagerCompat

    @JvmStatic
    val activityOptionsCompat = factory.activityOptionsCompat
}